src/rocksdb.rs

```diff
@@ -1000,6 +1024,20 @@ impl Writable for DB {
     fn single_delete_cf(&self, cf: &CFHandle, key: &[u8]) -> Result<(), String> {
         self.single_delete_cf_opt(cf, key, &WriteOptions::new())
     }
+
+    fn delete_range(&self, begin_key: &[u8], end_key: &[u8]) -> Result<(), String> {
+        let batch = WriteBatch::new();
```

Comment:
When using C ABI to create a `WriteBatch`, it will allocate the write batch on heap. When using C++ API, the write batch is allocated on stack however.